extensions/wikia/PhalanxII/classes/PhalanxStatsPager.class.php

```diff
@@ -45,26 +45,34 @@ function getPagingQueries() {
 	}
 
 	function formatRow( $row ) {
-		$type = implode( ", ", Phalanx::getTypeNames( $row->ps_blocker_hit ?? $row->ps_blocker_type ) );
+		// $row->ps_blocker_hit can be null or zero
+		$type = implode( ", ",
+			Phalanx::getTypeNames( empty( $row->ps_blocker_hit ) ? $row->ps_blocker_type : $row->ps_blocker_hit ) );
+
 		$username = $row->ps_blocked_user;
 		$timestamp = $this->getLanguage()->timeanddate( $row->ps_timestamp );
 		$oWiki = WikiFactory::getWikiByID( $row->ps_wiki_id );
 		$url = $row->ps_referrer ?? '';
 		$url = ( empty( $url ) && isset( $oWiki ) ) ? $oWiki->city_url : $url;
 
 		// SUS-184: Render usernames containing spaces correctly
-		$encUserName = str_replace(' ', '_', $username );
+		$encUserName = $this->replaceSpaces( $username );
 
-		$specialContributionsURL = GlobalTitle::newFromText( 'Contributions', NS_SPECIAL, $row->ps_wiki_id )->getFullURL();
+		$specialContributionsURL =
+			GlobalTitle::newFromText( 'Contributions', NS_SPECIAL, $row->ps_wiki_id )->getFullURL();
 
 		if ( !empty( $specialContributionsURL ) ) {
-			$username = '[' . $specialContributionsURL . '/' . $encUserName . ' ' . $username . ']';
+			$specialContributionsURL = "[$specialContributionsURL/$encUserName $encUserName]";
 		}
 
 		$html  = Html::openElement( 'li' );
-		$html .= $this->msg( 'phalanx-stats-row', $type, $username, $url, $timestamp )->parse();
+		$html .= $this->msg( 'phalanx-stats-row', $type, $specialContributionsURL, $url, $timestamp )->parse();
 		$html .= Html::closeElement( 'li' );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$html". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)